actionview/lib/action_view/template/handlers/erb.rb

```diff
@@ -138,7 +138,14 @@ def call(template)
         #
         #   <% cache notification.event do %> # => nil
         def resource_cache_call_pattern
-          /\A(?:<%#.*%>)*\s*<%\s*cache\(?\s*(\w+)[\s\)]/m
+          /\A
+            (?:<%\#.*%>)*                 # optional initial comment
+            \s*                           # followed by optional spaces or newlines
+            <%\s*cache[\(\s]              # followed by an ERB call to cache
+            \s*                           # followed by optional spaces or newlines
+            (?<resource_name>\w+)         # capture the cache call argument as :resource_name
+            [\s\)]                        # followed by a space or close paren
+          /xm
```

Comment:
I ran into an issue with the `#` char - in normal regexp you don't need to escape it, but in multi-line (`/x`) regexp you need to escape it as `\#` , because Ruby comments are allowed *inside* these multi-line regexps. That was confusing.